sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/_encode.py

```diff
@@ -765,13 +765,13 @@ def encode_filter_set(value):
         else:
             if isinstance(name, str):
                 name = name.encode("utf-8")  # type: ignore
-            try:
+            if len(data) == 2 and not isinstance(data, (str, bytes)):
```

Comment:
probably because I don't know much python but from a high level, I wouldn't understand why it must be (leng == 2) and not 1. or 3 or 4 or 5. May be worthwhile to add some documentation. lol.